typing_extensions/src_py3/typing_extensions.py

```diff
@@ -1755,3 +1772,165 @@ def get_type_hints(obj, globalns=None, localns=None, include_extras=False):
+
+    # Prior to python 3.7 types did not have `copy_with`. A lot of the equality
+    # checks, argument expansion etc.. are done on the _subs_tree.
+    #
+    # >> U = Union[Optional[float], int]
+    # >> U1 = Union[T, int]
+    # >> U2 = U1[Optional[float]]
+    # >> U == U2
+    # True
+    # >> U2.__args__, U.__args__
+    # (typing.Union[float, NoneType],) (<class 'float'>, <class 'NoneType'>, <class 'int'>)
+    #
+    # Here U and U2 have different shapes but have the same _subs_tree.
+    #
+    # As a result we can't provide a get_type_hints function that strips out
+    # annotations or a '__type__' field on the Annotated class.
+
+    class AnnotatedMeta(typing.GenericMeta):
+        """Metaclass for Annotated"""
+
+        def __new__(cls, name, bases, namespace, **kwargs):
+            if any(b is not object for b in bases):
+                raise TypeError("Cannot subclass " + str(Annotated))
+            return super().__new__(cls, name, bases, namespace, **kwargs)
+
+        @property
+        def __metadata__(self):
+            return self._subs_tree()[2]
+
+        def _tree_repr(self, tree):
+            assert len(tree) == 3
+            # First argument is this class, second is __type__, 3rd is __metadata__
+            tp_tree = tree[1]
+            if not isinstance(tp_tree, tuple):
+                tp_repr = typing._type_repr(tp_tree)
+            else:
+                tp_repr = tp_tree[0]._tree_repr(tp_tree)
+            metadata_reprs = ", ".join(repr(arg) for arg in tree[2])
+            return repr(tree[0]) + '[%s, %s]' % (tp_repr, metadata_reprs)
+
+        def _subs_tree(self, tvars=None, args=None):
+            if self is Annotated:
+                return Annotated
+            res = super()._subs_tree(tvars=tvars, args=args)
+            # Flatten nested Annotated
+            if isinstance(res[1], tuple) and res[1][0] is Annotated:
+                sub_tp = res[1][1]
+                sub_annot = res[1][2]
+                return (Annotated, sub_tp, sub_annot + res[2])
+            return res
+
+
+        def _get_cons(self, msg):
+            if self.__origin__ is None:
+                raise TypeError(msg + " a non specialized Annotated type")
+            tree = self._subs_tree()
+            while isinstance(tree, tuple) and tree[0] is Annotated:
+                tree = tree[1]
+            if isinstance(tree, tuple):
+                cons = tree[0]
+            else:
+                cons = tree
+            type_error = None
+            if isinstance(cons, typing._ForwardRef):
+                type_error = "a forward reference."
+            elif isinstance(cons, typing.TypeVar):
+                type_error = "a type variable."
+
+            if type_error:
+                raise TypeError(
+                    "{} {}: the annotated type is {}".format(
+                        msg,
+                        self,
+                        type_error
+                    )
+                )
```

Comment:
Please don't use this style, just put it all on the same line with string, it will be more readable.